pkg/util/container/ring/buffer.go

```diff
@@ -28,31 +44,49 @@ func NewBuffer(length int) *Buffer {
 	cv := sync.NewCond(m)
 
 	return &Buffer{
-		items:      make([][]byte, length),
+		items:      make([]Entry, length),
 		notEmpty:   cv,
 		mu:         m,
 		writeIndex: -1,
 	}
 }
 
 // Put an item on the ring buffer.
-func (r *Buffer) Put(item []byte) bool {
-	if item == nil || r.closed.Load() {
+func (r *Buffer) Put(i Entry) bool {
+	if i.Data == nil || r.closed.Load() {
 		return false
 	}
 
 	// Protect the slice and the writeIndex
 	r.mu.Lock()
 
-	if !r.Has(item) {
+	if !r.Has(i.Data) {
 		// Store the new item
 		r.writeIndex++
 		// Reset the writeIndex as this is ringBuffer
 		if int(r.writeIndex) == len(r.items) {
 			r.writeIndex = 0
 		}
 
-		r.items[r.writeIndex] = item
+		r.items[r.writeIndex] = i
+
+		// Simple sorting by priority. If most recent item is High Priority
+		// then search for any Low Priority message pushed before. If found, swap
+		// them.
+		if r.items[r.writeIndex].Priority == High {
+			for i := 0; i < len(r.items); i++ {
+				if r.items[i].Data == nil {
+					continue
+				}
+
+				if r.items[i].Priority == Low {
+					// found Low Priority message
+					// swap them
+					r.items[i], r.items[r.writeIndex] = r.items[r.writeIndex], r.items[i]
+					break
+				}
+			}
+		}
```

Comment:
Considering that the priority introduces swapping of `Entries` after a full scan of items, it is preferable to let `Entry` implement a `sort.Interface` and sort the items based on priority. This has the advantage to be more concise, likely more efficient and allow for an arbitrary number of priorities rather than two

Reply:
I don't think performance is really important here. It is more about readability and conciseness

Reply:
This has been addressed in #1086 PR